hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -62,25 +72,32 @@ public class MigrationRecordsManager {
 
 	private static final Key immutableKey = Key.newBuilder().setKeyList(KeyList.getDefaultInstance()).build();
 	private static final String MEMO = "Release 0.24.1 migration record";
+	private static final String CONTRACT_UPGRADE_MEMO = "Contract {} was renewed during 0.26.0 upgrade. New expiry: " +
+			"{}" +
+			" .";
 
 	private final EntityCreator creator;
 	private final SigImpactHistorian sigImpactHistorian;
 	private final RecordsHistorian recordsHistorian;
 	private final Supplier<MerkleNetworkContext> networkCtx;
 
 	private Supplier<SideEffectsTracker> sideEffectsFactory = SideEffectsTracker::new;
+	private SyntheticTxnFactory syntheticTxnFactory = new SyntheticTxnFactory();
```

Comment:
You can inject the `SyntheticTxnFactory` via the constructor, there's a singleton in the Dagger2 context.

Reply:
Got it !